### PR TITLE
feat: progress component

### PR DESCRIPTION
**Pull request by anhle10051996**
## Check List
- [x] Your code builds clean without any errors or warnings.
- [x] Check coding convention.
- [x] Updated Storybook components and stories to reflect any changes made to the UI.
- [x] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot 2023-04-13 at 11 19 02

https://user-images.githubusercontent.com/88309815/231654652-049e3d5f-8e63-4ea3-a568-839f166902a4.mp4

Screenshot 2023-07-07 at 09 41 50

### CODE REVIEW

File: src/components/Progress.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }, [progressWidth, value])
+
+    return (
+      <View
+        ref={ref}
+        style={StyleSheet.flatten([
+          styles.progress,
+          {width: width ? width : '100%', height: size, backgroundColor, borderRadius},
+        ])}
+        onLayout={({nativeEvent: {layout}}) => setProgressWidth(layout.width)}>
```
Comment (ngochuyduong) [inline src/components/Progress.tsx:94]: make separate function

File: src/components/Progress.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    )
+  },
+)
+
+Progress.displayName = 'Progress'
+
+export default memo(Progress)
+
+const styles = StyleSheet.create({
+  progress: {
```
Comment (ngochuyduong) [inline src/components/Progress.tsx:131]: use styled component

**Review by ngochuyduong — CHANGES_REQUESTED** [review]

File: src/components/Progress.tsx (hunk 1 above)
Comment (anhle10051996) [reply src/components/Progress.tsx]: Updated

File: src/components/Progress.tsx (hunk 2 above)
Comment (anhle10051996) [reply src/components/Progress.tsx]: Updated

**Comment by TranQuangPhi** [comment]
Please fill in the options on the checklist and try to upload the video again

**Comment by TranQuangPhi** [comment]
write test code for this component

**Comment by anhle10051996** [comment]
> Please fill in the options on the checklist and try to upload the video again

Updated

**Comment by anhle10051996** [comment]
> write test code for this component

updated

**Review by huydosgtech — APPROVED** [review]

File: src/components/Progress/Progress.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
+import React, {useEffect, useRef, useState, forwardRef, memo, useCallback} from 'react'
+import {View, Animated, LayoutChangeEvent} from 'react-native'
```
Comment (loc-nguyenthien) [inline src/components/Progress/Progress.tsx:2]: use `react-native-reanimated`

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/components/Progress/Progress.tsx (hunk 3 above)
Comment (anhle10051996) [reply src/components/Progress/Progress.tsx]: updated

File: src/components/Progress/Progress.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+const MAX_VALUE = 100
+
+const Progress = forwardRef<View, IProgressProps>(
+  (
+    {
+      width,
+      value = 0,
+      size = metrics.small,
+      borderRadius = 0,
+      filledTrackColor = '#49BE25',
```
Comment (ngochuyduong) [inline src/components/Progress/Progress.tsx:67]: try to use color from theme

**Comment by ThinhKimVo** [comment]
https://github.com/saigontechnology/rn-base-component/assets/104345392/e46f9c40-a2f2-4249-89d6-1f3ebce2174c

File: src/components/Progress/Progress.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+const MAX_VALUE = 100
+
+const Progress = forwardRef<View, IProgressProps>(
+  (
+    {
+      width,
+      value = 0,
+      size = metrics.small,
+      borderRadius = 0,
+      filledTrackColor = colors.primary,
```
Comment (ngochuyduong) [inline src/components/Progress/Progress.tsx:66]: use color from theme

**Review by baonguyenhsts — APPROVED** [review]

**Review by vunguyendhSTS — APPROVED** [review]

File: src/components/Progress/Progress.tsx (hunk 5 above)
Comment (ThinhKimVo) [reply src/components/Progress/Progress.tsx:66]: how to use theme on forwardRef props

Comment (ngochuyduong) [reply src/components/Progress/Progress.tsx:66]: color is import {colors} from '../../helpers/colors' 

import {theme} from '../../theme'
I mean we change to use color from the theme, ex: theme.colors.primary

**Review by vunguyendhSTS — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Review by ngochuyduong — APPROVED** [review]
